### PR TITLE
adding deployment file

### PR DESCRIPTION
**Pull request by johnoloughlin**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,6 +13,7 @@ help:
 	@echo "make build                 - Build UserSpace CNI container."
 	@echo "make deploy                - Copy binary from container to host: /opt/cni/bin."
 	@echo "make all                   - build and deploy"
+	@echo "make rollout               - kubectl apply daemonset"
```
Comment (Eoghan1232) [inline Makefile:16]: I'm not a fan of calling it rollout, I would prefer make deploy for something like this...

can we rename to deploy and choose another name for the other command?
"make copy" ?

File: kubernetes/userspace-daemonset.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        image: localhost:5000/userspacecni #registory:imagename
+        imagePullPolicy: IfNotPresent
+        volumeMounts:
+        - name: cnibin
+          mountPath: /opt/cni/bin
+      volumes:
+        - name: cnibin
+          hostPath:
+            path: /opt/cni/bin
+      containers:
```
Comment (Eoghan1232) [inline kubernetes/userspace-daemonset.yml:27]: so the init container copies the binary, and then it goes into the main container, which is a pause container, thus releasing resources.
correct?

File: Makefile (hunk 1 above)
Comment (johnoloughlin) [reply Makefile:16]: agree

File: Makefile [hunk 3]
```diff
@@ -23,8 +24,14 @@ deploy:
...
+	# Used in dockerfile
 	@cd userspace && go build -v
 
 generate:
+	# Used in dockerfile
 	for package in cnivpp/api/* ; do cd $$package ; pwd ; go generate ; cd - ; done
 
+rollout:
+	# Use sed to replace image name and then apply deploument file
+	sed "s|\(image:\).*\(#registory\)|\1 $(IMAGE_NAME) \2|g" ./kubernetes/userspace-daemonset.yml |kubectl apply -f -
```
Comment (Eoghan1232) [inline Makefile:36]: registry -> IMAGE_REGISTRY ?

Comment (Eoghan1232) [reply Makefile:36]: nvm, this check for "image: " and then for the end of sentence "#registry" and then adds the image_name in between them.

**Review by Eoghan1232 — APPROVED** [review]

**Pull request merged**